src/components/map/map-control.js

```diff
@@ -179,148 +179,201 @@ const MapControlPanel = React.memo(
+  const defaultActionIcons = {
+    cube() {
+      return <Cube3d height="22px" />;
+    }
+  };
+  const Toggle3dButton = React.memo(
+    ({dragRotate, onTogglePerspective, actionIcons = defaultActionIcons}) => (
+      <MapControlButton
+        onClick={e => {
+          e.preventDefault();
+          onTogglePerspective();
+        }}
+        active={dragRotate}
+        data-tip
+        data-for="action-3d"
+      >
+        {actionIcons.cube()}
+        <MapControlTooltip
+          id="action-3d"
+          message={dragRotate ? 'tooltip.disable3DMap' : 'tooltip.3DMap'}
+        />
+      </MapControlButton>
+    )
+  );
 
+  Toggle3dButton.displayName = 'Toggle3dButton';
+  return Toggle3dButton;
+}
 const StyledToolbar = styled(VerticalToolbar)`
   position: absolute;
   right: 32px;
 `;
 
-const MapDrawPanel = React.memo(
-  ({editor, isActive, onToggleMenuPanel, onSetEditorMode, onToggleEditorVisibility}) => {
-    return (
-      <div className="map-draw-controls" style={{position: 'relative'}}>
-        {isActive ? (
-          <StyledToolbar show={isActive}>
-            <ToolbarItem
-              className="edit-feature"
-              onClick={() => onSetEditorMode(EDITOR_MODES.EDIT)}
-              label="toolbar.select"
-              iconHeight="22px"
-              icon={CursorClick}
-              active={editor.mode === EDITOR_MODES.EDIT}
-            />
-            <ToolbarItem
-              className="draw-feature"
-              onClick={() => onSetEditorMode(EDITOR_MODES.DRAW_POLYGON)}
-              label="toolbar.polygon"
-              iconHeight="22px"
-              icon={Polygon}
-              active={editor.mode === EDITOR_MODES.DRAW_POLYGON}
-            />
-            <ToolbarItem
-              className="draw-rectangle"
-              onClick={() => onSetEditorMode(EDITOR_MODES.DRAW_RECTANGLE)}
-              label="toolbar.rectangle"
-              iconHeight="22px"
-              icon={Rectangle}
-              active={editor.mode === EDITOR_MODES.DRAW_RECTANGLE}
-            />
-            <ToolbarItem
-              className="toggle-features"
-              onClick={onToggleEditorVisibility}
-              label={editor.visible ? 'toolbar.hide' : 'toolbar.show'}
-              iconHeight="22px"
-              icon={editor.visible ? EyeSeen : EyeUnseen}
-            />
-          </StyledToolbar>
-        ) : null}
-        <MapControlButton
-          onClick={e => {
-            e.preventDefault();
-            onToggleMenuPanel();
-          }}
-          active={isActive}
-          data-tip
-          data-for="map-draw"
-        >
-          <DrawPolygon height="22px" />
-          <MapControlTooltip id="map-draw" message="tooltip.DrawOnMap" />
-        </MapControlButton>
-      </div>
-    );
-  }
-);
+MapDrawPanelFactory.deps = [];
+export function MapDrawPanelFactory() {
+  const defaultActionIcons = {
+    visible: EyeSeen,
+    hidden: EyeUnseen,
+    polygon() {
```

Comment:
each item should have the same construct, why is value of `visible` a component, and `polygon` a function?
if you are providing a prop API does it make sense for people to provide icons in different type?

```
polygon: () => <DrawPolygon height="22px" />
```

Reply:
you're correct. fixed